src/OrleansSQLUtils/Storage/Provider/RelationalStorageProvider.cs

```diff
@@ -0,0 +1,550 @@
+            if(string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("The parameter must contain characters", nameof(name));
+            }
+
+            if(providerRuntime == null)
+            {
+                throw new ArgumentNullException(nameof(providerRuntime));
+            }
+
+            if(config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            if(!config.Properties.ContainsKey(DataConnectionStringPropertyName))
+            {
+                throw new BadProviderConfigException($"The {DataConnectionStringPropertyName} setting has not been configured. Add a {DataConnectionStringPropertyName} setting with a valid connection string.");
+            }
+
+            //NOTE: StorageSerializationPicker should be defined outside and given as a parameter in constructor or via Init in IProviderConfiguration perhaps.
+            //Currently this limits one's options to much to the current situation of providing only one serializer for serialization and deserialization
+            //with no regard to state update or serializer changes. Maybe have this serialized as a JSON in props and read via a key?
+            StorageSerializationPicker = new DefaultRelationalStoragePicker(ConfigureDeserializers(config), ConfigureSerializers(config));
+
+            //NOTE: Currently there should be only one pair of providers given. That is, only UseJsonFormatPropertyName, UseXmlFormatPropertyName or UseBinaryFormatPropertyName.
+            if(StorageSerializationPicker.Deserializers.Count > 1 || StorageSerializationPicker.Serializers.Count > 1)
+            {
+                throw new ArgumentException("Configuration error, only one serializer and deserializer should be given.", nameof(config));
+            }
+
+            if(StorageSerializationPicker.Deserializers.Count == 0 || StorageSerializationPicker.Serializers.Count == 0)
+            {
+                StorageSerializationPicker.Deserializers.Add(new OrleansStorageDefaultBinaryDeserializer(UseBinaryFormatPropertyName));
+                StorageSerializationPicker.Serializers.Add(new OrleansStorageDefaultBinarySerializer(UseBinaryFormatPropertyName));
+            }
+
+            var connectionInvariant = config.Properties.ContainsKey(DataConnectionInvariantPropertyName) ? config.Properties[DataConnectionInvariantPropertyName] : DefaultAdoInvariantInvariantPropertyName;
+            Storage = RelationalStorage.CreateInstance(connectionInvariant, config.Properties[DataConnectionStringPropertyName]);
+            ServiceId = providerRuntime.ServiceId.ToString();
+
+            var queries = await Storage.ReadAsync(DefaultInitializationQuery, command => { }, (selector, resultSetCount, token) =>
+            {
+                return Task.FromResult(Tuple.Create(selector.GetValue<string>("QueryKey"), selector.GetValue<string>("QueryText")));
+            }).ConfigureAwait(false);
```

Comment:
To be fair, not that I would force everyone to use `ConfigureAwait(false)` in non-grain code, but I agree that it can be used when it is appropriate. For example, we are already [using it in other storage providers](https://github.com/dotnet/orleans/blob/master/src/OrleansAzureUtils/Providers/Storage/AzureBlobStorage.cs#L124), and I think we should start using it more when it's appropriate in more of Orleans internals.
The reality is that deadlocks arising from `Wait()` is just one of the nastiest side effects of using this incorrectly, and I agree any usage of this is very discouraged in Orleans anyway. Nevertheless, there's side effects in terms of scheduling and also unnecessary context propagation. They should be minor (and hence why I won't force everybody to start using ConfigureAwait all over the place, because it adds complexity if devs have to stop and think when does it apply), but why would this code have to wait to be scheduled honoring grain turns, if it's not grain code and doesn't care about reentrancy or interleaving?
Again, I wouldn't complain if we remove it from here, but looks like @veikkoeeva already went through the thought process and concluded the context is irrelevant here. Now it's up to both of you guys to decide, since @shayhatsor will be reviewing most of it anyway.